src/main/scala/with_monads/Writer.scala

```diff
@@ -1,19 +1,22 @@
 package with_monads
 
-trait WriterOps[A] {
-  def value(value: A): Writer[A]
-  def tell(entries: List[String]): Writer[A]
+import cats.Semigroup
+import cats.kernel.Monoid
+
+trait WriterOps[L, A] {
+  def value(value: A): Writer[L, A]
+  def tell(entries: L): Writer[L, A]
 }
 
-case class Writer[A](log: List[String], value: A) extends WriterOps[A] { 
-  def value(newValue: A): Writer[A]          = Writer(log, newValue)
-  def tell(entries: List[String]): Writer[A] = Writer(log ++ entries, value)
+case class Writer[L: Semigroup, A](log: L , value: A) extends WriterOps[L, A] { 
```

Comment:
The way this is written, we have two different requirements for `L`. `L` needs to be a `Semigroup` here, but it needs to be a `Monoid` in the definition of `monadForWriter`. The way you have it now compiles, but it causes a bit of strangeness for the user of the API, which I'll describe below.

To construct an instance of `Writer` using this implementation, we only need `L` to be a `Semigroup`. We can construct an instance of `Writer` using some `L` that's only a `Semigroup` (and not also a `Monoid`). We might not even notice anything is wrong if all we want to do is use `value` and `tell`.  But the moment we try and use something defined inside `Monad` (like `map` or `flatMap`) we'll run into a strange compiler error, telling us that there is no `Monad` available for `Writer.

The reason we'll see that error is because, when Scala tries to construct an implicit `Monad` instance, it looks for implicit evidence for `Monoid[L]`... and doesn't find it.

The issue is that this implementation requires a `Monoid` to construct the `Monad` instance, but requires a `Semigroup` to construct the `Writer` instance. We wouldn't even notice the error unless we test the code with something that's only a `Semigroup` in cats (e.g. a `NonEmptyList`).